biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceUtils.java

```diff
@@ -546,6 +546,8 @@ public static VersionedClause toVersionClause(Resource resource, String mask) {
 		if (resource.getCapabilities(WORKSPACE_NAMESPACE).isEmpty()) {
 			Macro macro = new Macro(new Processor());
 			Version version = getVersion(idCap);
+			if (version == null)
+				version = Version.emptyVersion;
```

Comment:
True. But they are specified to default to 0 when absent. The osgi.identity capability is specified that the version attribute is mandatory. The type attribute is also mandatory. If the indexed bundle has no Bundle-Version header, then the index generator must write out the osgi.identity version attribute with a 0 value.